{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<_csv.reader object at 0x7f19dc356c10>\n"
     ]
    }
   ],
   "source": [
    "import csv\n",
    "# python 读取文件\n",
    "with open(\"/root/yeast_data/data/RAN_seq/SraRunInfo.csv\") as f:\n",
    "    # python 读取csv\n",
    "    reader = csv.reader(f)\n",
    "    print(reader)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>ReleaseDate</th>\n",
       "      <th>LoadDate</th>\n",
       "      <th>spots</th>\n",
       "      <th>bases</th>\n",
       "      <th>spots_with_mates</th>\n",
       "      <th>avgLength</th>\n",
       "      <th>size_MB</th>\n",
       "      <th>AssemblyName</th>\n",
       "      <th>download_path</th>\n",
       "      <th>...</th>\n",
       "      <th>Affection_Status</th>\n",
       "      <th>Analyte_Type</th>\n",
       "      <th>Histological_Type</th>\n",
       "      <th>Body_Site</th>\n",
       "      <th>CenterName</th>\n",
       "      <th>Submission</th>\n",
       "      <th>dbgap_study_accession</th>\n",
       "      <th>Consent</th>\n",
       "      <th>RunHash</th>\n",
       "      <th>ReadHash</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR1916152</td>\n",
       "      <td>2015-04-03 17:06:06</td>\n",
       "      <td>2015-03-15 15:16:45</td>\n",
       "      <td>17028551</td>\n",
       "      <td>868456101</td>\n",
       "      <td>0</td>\n",
       "      <td>51</td>\n",
       "      <td>885</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GEO</td>\n",
       "      <td>SRA246757</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>AF738628B3B21AF0B2D5222743D803C2</td>\n",
       "      <td>608F2BCDA164CC23D1178526A9B1DD71</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR1916153</td>\n",
       "      <td>2015-04-03 17:06:06</td>\n",
       "      <td>2015-03-15 15:16:51</td>\n",
       "      <td>16542825</td>\n",
       "      <td>843684075</td>\n",
       "      <td>0</td>\n",
       "      <td>51</td>\n",
       "      <td>862</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GEO</td>\n",
       "      <td>SRA246757</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>354CD8BF3729EE6F6ED491F76DECC476</td>\n",
       "      <td>2C9F3C341FFDD3869E3740A6E75CF7AF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR1916154</td>\n",
       "      <td>2015-04-03 17:06:06</td>\n",
       "      <td>2015-03-15 15:16:39</td>\n",
       "      <td>16739897</td>\n",
       "      <td>853734747</td>\n",
       "      <td>0</td>\n",
       "      <td>51</td>\n",
       "      <td>874</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GEO</td>\n",
       "      <td>SRA246757</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>FE3A847B05887981D5F764D58F3138C4</td>\n",
       "      <td>954C10D5258D83193BCE73CD256D90E8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR1916155</td>\n",
       "      <td>2015-04-03 17:06:06</td>\n",
       "      <td>2015-03-15 15:16:33</td>\n",
       "      <td>16534790</td>\n",
       "      <td>843274290</td>\n",
       "      <td>0</td>\n",
       "      <td>51</td>\n",
       "      <td>862</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GEO</td>\n",
       "      <td>SRA246757</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>CC44A1715233D6FE50FCF1B9CF8FDE14</td>\n",
       "      <td>5EA08B9FDB5359DCB4E25162EE8A12AC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR1916156</td>\n",
       "      <td>2015-04-03 17:06:06</td>\n",
       "      <td>2015-03-15 15:16:39</td>\n",
       "      <td>15793246</td>\n",
       "      <td>805455546</td>\n",
       "      <td>0</td>\n",
       "      <td>51</td>\n",
       "      <td>827</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GEO</td>\n",
       "      <td>SRA246757</td>\n",
       "      <td>NaN</td>\n",
       "      <td>public</td>\n",
       "      <td>396A87E0D6BDE9A0F5AAA11C0D6D95BB</td>\n",
       "      <td>00A7ADB71C6CF1654686C60F425DB897</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 47 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Run          ReleaseDate             LoadDate     spots      bases  \\\n",
       "0  SRR1916152  2015-04-03 17:06:06  2015-03-15 15:16:45  17028551  868456101   \n",
       "1  SRR1916153  2015-04-03 17:06:06  2015-03-15 15:16:51  16542825  843684075   \n",
       "2  SRR1916154  2015-04-03 17:06:06  2015-03-15 15:16:39  16739897  853734747   \n",
       "3  SRR1916155  2015-04-03 17:06:06  2015-03-15 15:16:33  16534790  843274290   \n",
       "4  SRR1916156  2015-04-03 17:06:06  2015-03-15 15:16:39  15793246  805455546   \n",
       "\n",
       "   spots_with_mates  avgLength  size_MB  AssemblyName  \\\n",
       "0                 0         51      885           NaN   \n",
       "1                 0         51      862           NaN   \n",
       "2                 0         51      874           NaN   \n",
       "3                 0         51      862           NaN   \n",
       "4                 0         51      827           NaN   \n",
       "\n",
       "                                       download_path  ... Affection_Status  \\\n",
       "0  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...  ...              NaN   \n",
       "1  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...  ...              NaN   \n",
       "2  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...  ...              NaN   \n",
       "3  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...  ...              NaN   \n",
       "4  https://sra-downloadb.be-md.ncbi.nlm.nih.gov/s...  ...              NaN   \n",
       "\n",
       "   Analyte_Type Histological_Type Body_Site CenterName Submission  \\\n",
       "0           NaN               NaN       NaN        GEO  SRA246757   \n",
       "1           NaN               NaN       NaN        GEO  SRA246757   \n",
       "2           NaN               NaN       NaN        GEO  SRA246757   \n",
       "3           NaN               NaN       NaN        GEO  SRA246757   \n",
       "4           NaN               NaN       NaN        GEO  SRA246757   \n",
       "\n",
       "   dbgap_study_accession  Consent                           RunHash  \\\n",
       "0                    NaN   public  AF738628B3B21AF0B2D5222743D803C2   \n",
       "1                    NaN   public  354CD8BF3729EE6F6ED491F76DECC476   \n",
       "2                    NaN   public  FE3A847B05887981D5F764D58F3138C4   \n",
       "3                    NaN   public  CC44A1715233D6FE50FCF1B9CF8FDE14   \n",
       "4                    NaN   public  396A87E0D6BDE9A0F5AAA11C0D6D95BB   \n",
       "\n",
       "                           ReadHash  \n",
       "0  608F2BCDA164CC23D1178526A9B1DD71  \n",
       "1  2C9F3C341FFDD3869E3740A6E75CF7AF  \n",
       "2  954C10D5258D83193BCE73CD256D90E8  \n",
       "3  5EA08B9FDB5359DCB4E25162EE8A12AC  \n",
       "4  00A7ADB71C6CF1654686C60F425DB897  \n",
       "\n",
       "[5 rows x 47 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "df = pd.read_csv(\"/root/yeast_data/data/RAN_seq/SraRunInfo.csv\")\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "fastq-dump SRR1916152 \n",
      "fastqc SRR1916152.fastq\n",
      "fastq-dump SRR1916153 \n",
      "fastqc SRR1916153.fastq\n",
      "fastq-dump SRR1916154 \n",
      "fastqc SRR1916154.fastq\n",
      "fastq-dump SRR1916155 \n",
      "fastqc SRR1916155.fastq\n",
      "fastq-dump SRR1916156 \n",
      "fastqc SRR1916156.fastq\n"
     ]
    }
   ],
   "source": [
    "for index,row in df.iterrows():\n",
    "    cmd_str = \"fastq-dump {0} \\nfastqc {0}.fastq\".format(row[0])\n",
    "    print(cmd_str)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### hisat2将reads比对到参考基因组"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hisat2 -x ../../data/reference/yeast_ref -U ../../data/RAN_seq/SRR1916152.fastq | samtools view -bS - | samtools sort - -o EV_3.bam\n",
      "hisat2 -x ../../data/reference/yeast_ref -U ../../data/RAN_seq/SRR1916153.fastq | samtools view -bS - | samtools sort - -o EV_4.bam\n",
      "hisat2 -x ../../data/reference/yeast_ref -U ../../data/RAN_seq/SRR1916154.fastq | samtools view -bS - | samtools sort - -o DNMT3B_2.bam\n",
      "hisat2 -x ../../data/reference/yeast_ref -U ../../data/RAN_seq/SRR1916155.fastq | samtools view -bS - | samtools sort - -o DNMT3B_3.bam\n",
      "hisat2 -x ../../data/reference/yeast_ref -U ../../data/RAN_seq/SRR1916156.fastq | samtools view -bS - | samtools sort - -o DNMT3B_4.bam\n"
     ]
    }
   ],
   "source": [
    "with open(\"/root/yeast_data/data/RAN_seq/SraAccList.txt\") as acc_file_handle:\n",
    "    for line in acc_file_handle:\n",
    "        line = line.strip()                                                       \n",
    "        line = line.split(\"\\t\")\n",
    "        cmd_str=\"hisat2 -x ../../data/reference/yeast_ref -U ../../data/RAN_seq/{0}.fastq | samtools view -bS - | samtools sort - -o {1}.bam\".format(line[0],line[1])\n",
    "        #cmd_str+=\"\\nsamtools index {1}.bam\n",
    "        print(cmd_str)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### read count\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "htseq-count -f bam -r pos ../read_aln/EV_3.bam ../../data/reference/Saccharomyces_cerevisiae.R64-1-1.48.gtf > EV_3.count.tab\n",
      "htseq-count -f bam -r pos ../read_aln/EV_4.bam ../../data/reference/Saccharomyces_cerevisiae.R64-1-1.48.gtf > EV_4.count.tab\n",
      "htseq-count -f bam -r pos ../read_aln/DNMT3B_2.bam ../../data/reference/Saccharomyces_cerevisiae.R64-1-1.48.gtf > DNMT3B_2.count.tab\n",
      "htseq-count -f bam -r pos ../read_aln/DNMT3B_3.bam ../../data/reference/Saccharomyces_cerevisiae.R64-1-1.48.gtf > DNMT3B_3.count.tab\n",
      "htseq-count -f bam -r pos ../read_aln/DNMT3B_4.bam ../../data/reference/Saccharomyces_cerevisiae.R64-1-1.48.gtf > DNMT3B_4.count.tab\n"
     ]
    }
   ],
   "source": [
    "with open(\"/root/yeast_data/data/RAN_seq/SraAccList.txt\") as acc_file_handle:\n",
    "    for line in acc_file_handle:\n",
    "        line = line.strip()                                                       \n",
    "        line = line.split(\"\\t\")\n",
    "        cmd_str=\"htseq-count -f bam -r pos ../read_aln/{1}.bam ../../data/reference/Saccharomyces_cerevisiae.R64-1-1.48.gtf > {1}.count.tab\".format(line[0],line[1])\n",
    "        #cmd_str+=\"\\nsamtools index {1}.bam\n",
    "        \n",
    "        print(cmd_str)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['DNMT3B_2.count.tab', 'EV_3.count.tab', 'DNMT3B_3.count.tab', 'DNMT3B_4.count.tab', 'EV_4.count.tab']\n",
      "gene_id\tDNMT3B_2\tEV_3\tDNMT3B_3\tDNMT3B_4\tEV_4\n",
      "ETS1-1\t8\t14\t4\t7\t8\n",
      "ETS1-2\t0\t0\t0\t0\t0\n",
      "ETS2-1\t0\t0\t0\t1\t0\n",
      "ETS2-2\t0\t0\t0\t0\t0\n",
      "HRA1\t19\t71\t24\t14\t30\n",
      "ICR1\t203\t142\t210\t188\t159\n",
      "IRT1\t618\t387\t555\t491\t260\n",
      "ITS1-1\t7\t17\t4\t6\t\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "base_dir = \"/root/yeast_data/analysis/htseq-count\"\n",
    "dir_list = os.listdir(base_dir)\n",
    "file_name_list = []\n",
    "for i in dir_list:\n",
    "    if os.path.splitext(i)[1] == '.tab':\n",
    "        file_name_list.append(i)\n",
    "\n",
    "print(file_name_list)\n",
    "count_dict = {}\n",
    "for file_name in file_name_list:\n",
    "    with open(base_dir+\"/\"+file_name) as f:\n",
    "        for line in f:\n",
    "            if line.startswith(\"__\"):\n",
    "                continue\n",
    "            line = line.rstrip(\"\\n\")\n",
    "            ele = line.split()\n",
    "            if ele[0] in count_dict.keys():\n",
    "                count_dict.get(ele[0]).append(ele[1])\n",
    "            else:\n",
    "                count_dict[ele[0]]=[ele[1]]\n",
    "# print(count_dict)\n",
    "result_str=\"\"\n",
    "#print(\"gene_id\\t\" + \"\\t\".join([x[:-10] for x in file_name_list]))\n",
    "result_str +=\"gene_id\\t\" + \"\\t\".join([x[:-10] for x in file_name_list])+\"\\n\"\n",
    "for gene_id in count_dict:\n",
    "    #print(gene_id+\"\\t\"+\"\\t\".join(count_dict[gene_id]))\n",
    "    result_str+= gene_id+\"\\t\"+\"\\t\".join(count_dict[gene_id])+\"\\n\"\n",
    "\n",
    "print(result_str[:200])\n",
    "with open(base_dir+\"/result.txt\",\"w\") as f:\n",
    "    f.write(result_str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
